### PR TITLE
feat: New rule - SingleShapePointValidator for flagging shapes with a single shape point

### PR DESCRIPTION
**Pull request by praneethd7**
**Summary:**

Reference - #1733.  Some GTFS feeds have shapes with a single shape point. These route shapes cannot be visualized on any GIS software (which requires LineString) and are therefore indicative of an error in the creation (as a line requires a minimum of two points)

**Expected behavior:** 

The new rule flags such cases with the `single_shape_point` notice which reports the `shape_id` and `csvRowNumber` of the shape with a single shape point(one row in the feed).

Example Feed - LANTA GTFS:
image

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Add or update any needed [documentation](https://github.com/MobilityData/gtfs-validator/tree/master/docs) to the repo 
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [x] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

**Comment by qcdyx** [comment]
Hello @praneethd7, thank you for contributing to the GTFS Validator. I have two pieces of feedback:
1. The Java Formatting check is failing. Could you fix it by running ./gradlew goJF or gradle goJF?
2. The Rule acceptance test/compare-outputs check is failing. There are 22 new warnings out of 1520 datasets (~1%) that are invalid due to a code change, which exceeds the 1% threshold. Could you review these 22 datasets to see if they actually have the single shape point issue?

**Comment by praneethd7** [comment]
Hi @qcdyx! Thank you for your suggestions. I have re-formatted the code according to the Google Java format. Is there a way to know which are those 22 feeds that have this error? Also, I am not sure if I have access to all of them. Please let me know.😄

**Comment by qcdyx** [comment]
Hello @praneethd7 Yes, you have access to those 22 feeds. To get them, goto https://github.com/MobilityData/gtfs-validator/actions/runs/9197621511?pr=1753, scroll down to bottom, find the [acceptance_test_report] artefact(https://github.com/MobilityData/gtfs-validator/actions/runs/9197621511/artifacts/1541824946), and download it. Unzip acceptance_test_report.zip, open acceptance_test_report.json and search for "sourceUrl", you can click the link and download the dataset.
Don't forget to format code again since we merge master branch into your branch.

**Comment by praneethd7** [comment]
@qcdyx Thank you for helping me with the code formatting. I have manually checked the 22 feeds and they all have `single_shape_point` error. I have verified the notice counts for each feed and everything seems to be reported correctly. I guess we are good to go :)

**Comment by qcdyx** [comment]
@emmambd Should we consult with the spec team to verify the naming and description of the single_shape_point notice?

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/SingleShapePointValidator.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            new SingleShapePointNotice(
+                entry.getKey(), shapePointRowNumbers.getOrDefault(entry.getKey(), 0)));
+      }
+    }
+  }
+
+  /**
+   * The shape within `shapes.txt` contains a single shape point.
+   *
+   * <p>A shape should contain more than one shape point to visualize the route on any GIS software
```
Comment (emmambd) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/SingleShapePointValidator.java:52]: Request to modify description to "A shape should contain more than one shape point to visualize the route" to be more generic.

**Comment by emmambd** [comment]
@qcdyx No, that one's for me to review! @praneethd7 The name looks great! I made a minor comment on the description for the sake of being generic.

**Review by qcdyx — APPROVED** [review]
LGTM - ready to be merged after the modification of rule description

**Comment by praneethd7** [comment]
@emmambd @qcdyx Thank you for the suggestions. Changed the rule description

**Comment by qcdyx** [comment]
@praneethd7 Great, you PR is merged. Thanks again for contributing to GTFS Validator.

**Pull request merged**